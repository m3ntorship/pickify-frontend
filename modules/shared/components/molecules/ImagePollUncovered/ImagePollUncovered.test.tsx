import React from 'react';
import renderer from 'react-test-renderer';
import ImagePollUncoveredStory from './ImagePollUncovered';

describe('ImagePollUncoveredStory', () => {
  it('should render vertical Progress with primary meterColor when passing mostVoted=[true] leastVoted=[false]', () => {
    const tree = renderer.create(
      <ImagePollUncoveredStory
        type="vertical"
        mostVoted
        leastVoted={false}
        verticalMeterHeight={300}
        id="option_1"
        percentage={50}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render vertical Progress with red meterColor when passing mostVoted=[false] leastVoted=[true]', () => {
    const tree = renderer.create(
      <ImagePollUncoveredStory
        type="vertical"
        mostVoted={false}
        leastVoted
        verticalMeterHeight={300}
        id="option_1"
        percentage={50}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render vertical Progress with primary-shd05 meterColor when passing mostVoted=[false] leastVoted=[false]', () => {
    const tree = renderer.create(
      <ImagePollUncoveredStory
        type="vertical"
        mostVoted={false}
        leastVoted={false}
        verticalMeterHeight={300}
        id="option_1"
        percentage={50}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render circular Progress with primary meterColor when passing mostVoted=[true] leastVoted=[false]', () => {
    const tree = renderer.create(
      <ImagePollUncoveredStory
        type="circular"
        mostVoted
        leastVoted={false}
        id="option_1"
        percentage={50}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render circular Progress with red meterColor when passing mostVoted=[false] leastVoted=[true]', () => {
    const tree = renderer.create(
      <ImagePollUncoveredStory
        type="circular"
        mostVoted={false}
        leastVoted
        id="option_1"
        percentage={50}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
