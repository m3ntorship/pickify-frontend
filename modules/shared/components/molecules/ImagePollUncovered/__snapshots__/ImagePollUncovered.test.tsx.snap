// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePollUncoveredStory should render circular Progress with primary meterColor when passing mostVoted=[true] leastVoted=[false] 1`] = `
<div
  className="circular-vote-meter-wrapper"
  id="option_1"
>
  <div
    className="circular-content"
  >
    <p
      className="mb-1"
    >
      üëç
    </p>
    <p>
      50
      %
    </p>
  </div>
  <div
    className="absolute "
  >
    <svg
      height={100}
      width={100}
    >
      <circle
        className="circular primary-text"
        cx={50}
        cy={50}
        fill="transparent"
        r={36}
        strokeDasharray="226.1946710584651 226.1946710584651"
        strokeWidth={7}
        style={
          Object {
            "strokeDashoffset": "113.09733552923255",
          }
        }
      />
    </svg>
  </div>
</div>
`;

exports[`ImagePollUncoveredStory should render circular Progress with red meterColor when passing mostVoted=[false] leastVoted=[true] 1`] = `
<div
  className="circular-vote-meter-wrapper"
  id="option_1"
>
  <div
    className="circular-content"
  >
    <p
      className="mb-1"
    >
      üëé
    </p>
    <p>
      50
      %
    </p>
  </div>
  <div
    className="absolute "
  >
    <svg
      height={100}
      width={100}
    >
      <circle
        className="circular error-text"
        cx={50}
        cy={50}
        fill="transparent"
        r={36}
        strokeDasharray="226.1946710584651 226.1946710584651"
        strokeWidth={7}
        style={
          Object {
            "strokeDashoffset": "113.09733552923255",
          }
        }
      />
    </svg>
  </div>
</div>
`;

exports[`ImagePollUncoveredStory should render vertical Progress with primary meterColor when passing mostVoted=[true] leastVoted=[false] 1`] = `
<div
  className="vertical-vote-meter-wrapper"
  id="option_1"
>
  <div
    className="percentage-wrapper"
  >
    <p
      className="percentage"
    >
      üåü
       
      <span>
        50
        %
      </span>
    </p>
  </div>
  <div
    className="flex h-full"
  >
    <div
      className="outer-vertical wrapper"
      style={
        Object {
          "height": "150px",
        }
      }
    >
      <div
        className="inner-vertical primary"
      />
    </div>
  </div>
</div>
`;

exports[`ImagePollUncoveredStory should render vertical Progress with primary-shd05 meterColor when passing mostVoted=[false] leastVoted=[false] 1`] = `
<div
  className="vertical-vote-meter-wrapper"
  id="option_1"
>
  <div
    className="percentage-wrapper"
  >
    <p
      className="percentage"
    >
       
      <span>
        50
        %
      </span>
    </p>
  </div>
  <div
    className="flex h-full"
  >
    <div
      className="outer-vertical wrapper"
      style={
        Object {
          "height": "150px",
        }
      }
    >
      <div
        className="inner-vertical primary-shd5"
      />
    </div>
  </div>
</div>
`;

exports[`ImagePollUncoveredStory should render vertical Progress with red meterColor when passing mostVoted=[false] leastVoted=[true] 1`] = `
<div
  className="vertical-vote-meter-wrapper"
  id="option_1"
>
  <div
    className="percentage-wrapper"
  >
    <p
      className="percentage"
    >
       
      <span>
        50
        %
      </span>
    </p>
  </div>
  <div
    className="flex h-full"
  >
    <div
      className="outer-vertical wrapper"
      style={
        Object {
          "height": "150px",
        }
      }
    >
      <div
        className="inner-vertical error"
      />
    </div>
  </div>
</div>
`;
